sbt-coursier/src/main/scala-2.10/coursier/ToSbt.scala

```diff
@@ -1,90 +1,103 @@
 package coursier
 
 import java.util.GregorianCalendar
+import java.util.concurrent.ConcurrentHashMap
 
 import coursier.maven.MavenSource
-
 import sbt._
 
 object ToSbt {
 
-  def moduleId(dependency: Dependency): sbt.ModuleID =
+  private def caching[K, V](f: K => V): K => V = {
+
+    val cache = new ConcurrentHashMap[K, V]
```

Comment:
Note that the issue with `getOrElseUpdate` was fixed in 2.11.x. https://issues.scala-lang.org/browse/SI-7943